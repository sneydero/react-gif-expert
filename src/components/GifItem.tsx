import React from "react";

interface Props {
  title: string;
  url: string;
}

const GifItem = ({ title, url }: Props) => {
  return (
    <div className="card">
      <img src={url} alt={url} />
      <p>{title}</p>
    </div>
  );
};

export default GifItem;
