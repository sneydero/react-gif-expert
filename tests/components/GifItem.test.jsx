import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe("Test in <GifItem/>", () => {
  const title = "Esto es una prueba";
  const url = "http://image.test.com/";

  test("It should do match with snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("It should show the image and ALT with the url indicated", () => {
    render(<GifItem title={title} url={url} />);

    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(url);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(url);
  });

  test("It should show the title as a text in the page", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title).innerHTML).toBe(title);
  });
});
