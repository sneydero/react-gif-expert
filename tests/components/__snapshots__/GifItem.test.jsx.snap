// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> It should do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="http://image.test.com/"
      src="http://image.test.com/"
    />
    <p>
      Esto es una prueba
    </p>
  </div>
</div>
`;
